packages/gatsby-plugin-netlify-cms/src/gatsby-node.js

```diff
@@ -56,115 +56,122 @@ exports.onCreateWebpackConfig = (
     htmlTitle = `Content Manager`,
     manualInit = false,
   }
-) => {
-  if ([`develop`, `build-javascript`].includes(stage)) {
-    const gatsbyConfig = getConfig()
-    const { program } = store.getState()
-    const publicPathClean = trim(publicPath, `/`)
-    const config = {
-      ...gatsbyConfig,
-      entry: {
-        cms: [
-          manualInit && `${__dirname}/cms-manual-init.js`,
-          `${__dirname}/cms.js`,
-          enableIdentityWidget && `${__dirname}/cms-identity.js`,
-        ]
-          .concat(modulePath)
-          .filter(p => p),
-      },
-      output: {
-        path: path.join(program.directory, `public`, publicPathClean),
-      },
-      module: {
-        /**
-         * Manually swap `style-loader` for `MiniCssExtractPlugin.loader`.
-         * `style-loader` is only used in development, and doesn't allow us to
-         * pass the `styles` entry css path to Netlify CMS.
-         */
-        rules: deepMap(gatsbyConfig.module.rules, value => {
-          if (
-            typeof get(value, `loader`) === `string` &&
-            value.loader.includes(`style-loader`)
-          ) {
-            return { ...value, loader: MiniCssExtractPlugin.loader }
-          }
-          return value
-        }),
-      },
-      plugins: [
-        /**
-         * Remove plugins that either attempt to process the core Netlify CMS
-         * application, or that we want to replace with our own instance.
-         */
-        ...gatsbyConfig.plugins.filter(
-          plugin =>
-            ![`MiniCssExtractPlugin`, `GatsbyWebpackStatsExtractor`].find(
-              pluginName =>
-                plugin.constructor && plugin.constructor.name === pluginName
-            )
-        ),
+) =>
+  new Promise((resolve, reject) => {
+    if ([`develop`, `build-javascript`].includes(stage)) {
```

Comment:
If another stage is introduced that we do not want then this would break the build or may introduce in a multiple build. This is changing the way the build would work.